streams/src/main/java/org/apache/kafka/streams/processor/internals/StandbyTask.java

```diff
@@ -166,29 +169,70 @@ public void closeDirty() {
      * @throws TaskMigratedException all the task has been migrated
      * @throws StreamsException fatal error, should close the thread
      */
-    private void close(final boolean clean) {
+    private void prepareClose(final boolean clean) {
         if (state() == State.CREATED) {
             // the task is created and not initialized, do nothing
-            transitionTo(State.CLOSING);
-        } else {
-            if (state() == State.RUNNING) {
-                if (clean) {
-                    commit();
-                }
+            return;
+        }
 
-                transitionTo(State.CLOSING);
+        if (state() == State.RUNNING) {
+            if (clean) {
+                stateMgr.flush();
             }
+        } else {
+            throw new IllegalStateException("Illegal state " + state() + " while closing standby task " + id);
+        }
+    }
+
+    @Override
+    public void closeClean(final Map<TopicPartition, Long> checkpoint) {
+        Objects.requireNonNull(checkpoint);
+        close(true, checkpoint);
+
+        log.info("Closed clean");
+    }
+
+    @Override
+    public void closeDirty() {
+        close(false, null);
+
+        log.info("Closed dirty");
+    }
 
-            if (state() == State.CLOSING) {
-                executeAndMaybeSwallow(clean, () -> {
-                    StateManagerUtil.closeStateManager(log, logPrefix, clean,
-                        false, stateMgr, stateDirectory, TaskType.STANDBY);
-                }, "state manager close", log);
+    private void close(final boolean clean,
+                       final Map<TopicPartition, Long> checkpoint) {
+        if (state() == State.CREATED) {
+            // the task is created and not initialized, do nothing
+            closeTaskSensor.record();
```

Comment:
Well, we can, but we get an empty "than block" what is weird:
```
if (state() == State.CREATED) {
  // empty
} else if {
...
}
```

Reply:
I'd say we can always log a debug there saying "doing nothing in this function since we are in this state" :) The main concern I had, is that if in the future we want to add more steps in addition to `recording sensor` etc, we may forget adding it in one place or the other. Removing duplicated code helps us to be less vulnerable.